### DISCUSSION TITLE
EEprom help

### DISCUSSION THREAD

**Discussion by MYCAMEL222** [earlephilhower/arduino-pico#484] [Q&A] [answered]
I need to be able to write a number like 235876 to eeprom and and bring it back on a reboot.

I think I need to break the number up into 23 58 76 and write those into 3 consecutive addresses. Then read them back and put them back together by something like ..........230000+5800+76

Tell me there is an easier way please!

Thanks.

**Comment by Andy2No** [comment]
Break the number into bytes, using shifts and bit masks.  E.g.

```
// Pull apart
int num = 23456;
int lsb = num & 0xff; // lsb now contains just the lower 8 bits ("least significant" bits or byte) 
int mid = (num >> 8) & 0xff;  // middle 8 bits
int msb = (num >> 16) & 0xff;  // top 8 bits ("most significant" bits or byte)
  // The  & 0xff  part isn't really needed for the last one, if you're sure the number has no more than 24 bits

// Reassemble

num_rebuilt = (msb << 16) + (mid << 8) + lsb;
```

**Comment by MYCAMEL222** [comment, marked as answer]
Thanks!

Looks easier than what I was trying to do...
